overlay.d/05rhcos/usr/libexec/coreos-update-bootloader

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+set -euo pipefail
+
+# https://issues.redhat.com/browse/OCPBUGS-54594
+
+# This script updates the bootloader using bootupd when it's safe
+# and manually otherwise. Right now bootupd doesn't support RAID-1
+# setups and also not sure of the behavior if there are multiple
+# EFI-SYSTEM labeled filesystems attached.
+
+# Function that actually does the manual copy
+copy_to_esp_device() {
+    local device=$1
+    mount $device /boot/efi
+    echo "[Before Update: ${device}]"
+    find /boot/efi/ -type f | xargs md5sum
+    cp -rp /usr/lib/bootupd/updates/EFI /boot/efi
+    echo "[After Update: ${device}]"
+    find /boot/efi/ -type f | xargs md5sum
+    umount /boot/efi
```

Comment:
How about `umount -l /boot/efi` ? Might make sure the cp command finished? I remembered hit an error when doing tests with bootupd if using `umount` without `-l`.

Reply:
The only problem with using a lazy unmount is that we use the same mount point again immediately when we mount the next ESP for the copy. I don't know what the behavior is if the previous unmount hasn't succeeded yet.

Reply:
Good question, I do not know either. Agree to keep it as it is and see if stuck in problem. Maybe can add kola test for this?